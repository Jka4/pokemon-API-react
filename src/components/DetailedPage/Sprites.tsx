import React, { Suspense } from "react";
import Paper from "@material-ui/core/Paper";
import Skeleton, { SkeletonTheme } from 'react-loading-skeleton';

type Props = {
	sprites: any;
};

const Sprites: React.FC<Props> = ({ sprites = {} }: Props) => {

	const SpritesPlaceholders = () => {
		return (<>
			{[1, 2, 3, 4].map((el) => (
				<Paper key={el} elevation={3} className="block">
					<SkeletonTheme color="#6cff79" highlightColor="yellow">
						<Skeleton className="solo" circle={true} height={60} width={60} />
					</SkeletonTheme>
				</Paper>
			))}
		</>);
	};

	return (<>
		<div className="imagesLine">

			<Suspense fallback={SpritesPlaceholders()} >
				{Object.keys(sprites).map((spriteName: string) => (
					<div key={spriteName}>
						{sprites[spriteName] && (
							<Paper elevation={3} className="block">
								<img src={sprites[spriteName]} alt={sprites[spriteName]} className={sprites[spriteName]} />
							</Paper>
						)}
					</div>
				))}
			</Suspense>

		</div>
	</>);
};

export default Sprites;
