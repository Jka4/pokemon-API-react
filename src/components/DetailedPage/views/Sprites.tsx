import React from 'react';

import { spritesType } from 'types/index';
import styled from 'styled-components';

interface Props {
  sprites: spritesType;
}

const Sprites: React.FC<Props> = ({ sprites }: Props) => {
  return (
    <>
      <Images>
        {Object.keys(sprites).map((spriteName: string) => (
          <div key={spriteName}>
            {sprites[spriteName] && (
              <Block>
                <img src={sprites[spriteName]} alt={spriteName} />
              </Block>
            )}
          </div>
        ))}
      </Images>
    </>
  );
};

const Images = styled.div`
  margin-bottom: 100px;
  margin-top: 2rem;
  width: 100%;
  height: auto;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
`;

const Block = styled.div`
  min-width: 70px;
  margin: 10px;
  width: 100px;
  height: 100px;
  display: flex;
  border-radius: 5px;
  background: #6cff79;
  box-shadow: 3px 3px -2px rgb(0 0 0 / 20%), 0px 3px 4px 0px rgb(0 0 0 / 14%), 0px 1px 8px 0px rgb(0 0 0 / 12%);
`;

export default Sprites;
