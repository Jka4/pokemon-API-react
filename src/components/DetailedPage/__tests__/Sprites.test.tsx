import Sprites from '../views/Sprites';

import { mount } from 'enzyme';

describe('<Sprites/> ', () => {
  const fakeData = {
    back_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/168.png',
    back_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/168.png',
    front_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/168.png',
    front_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/168.png',
  };

  const container = mount(<Sprites sprites={fakeData} />);

  it('renders correctly', () => {
    expect(container).toMatchSnapshot();
  });
});
