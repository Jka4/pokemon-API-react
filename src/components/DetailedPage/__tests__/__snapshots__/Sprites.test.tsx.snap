// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sprites/>  renders correctly 1`] = `
<Sprites
  sprites={
    Object {
      "back_default": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/168.png",
      "back_shiny": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/168.png",
      "front_default": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/168.png",
      "front_shiny": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/168.png",
    }
  }
>
  <styled.div>
    <div
      className="sc-bdfBwQ kWmHqk"
    >
      <Suspense
        fallback={
          <React.Fragment>
            <ForwardRef(styled.div)>
              <SkeletonTheme
                color="#6cff79"
                highlightColor="yellow"
              >
                <Skeleton
                  circle={true}
                  count={1}
                  duration={1.2}
                  height={60}
                  width={60}
                  wrapper={null}
                />
              </SkeletonTheme>
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              <SkeletonTheme
                color="#6cff79"
                highlightColor="yellow"
              >
                <Skeleton
                  circle={true}
                  count={1}
                  duration={1.2}
                  height={60}
                  width={60}
                  wrapper={null}
                />
              </SkeletonTheme>
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              <SkeletonTheme
                color="#6cff79"
                highlightColor="yellow"
              >
                <Skeleton
                  circle={true}
                  count={1}
                  duration={1.2}
                  height={60}
                  width={60}
                  wrapper={null}
                />
              </SkeletonTheme>
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              <SkeletonTheme
                color="#6cff79"
                highlightColor="yellow"
              >
                <Skeleton
                  circle={true}
                  count={1}
                  duration={1.2}
                  height={60}
                  width={60}
                  wrapper={null}
                />
              </SkeletonTheme>
            </ForwardRef(styled.div)>
          </React.Fragment>
        }
      >
        <div
          key="back_default"
        >
          <styled.div>
            <div
              className="sc-gsTCUz dxnSOi"
            >
              <img
                alt="back_default"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/168.png"
              />
            </div>
          </styled.div>
        </div>
        <div
          key="back_shiny"
        >
          <styled.div>
            <div
              className="sc-gsTCUz dxnSOi"
            >
              <img
                alt="back_shiny"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/168.png"
              />
            </div>
          </styled.div>
        </div>
        <div
          key="front_default"
        >
          <styled.div>
            <div
              className="sc-gsTCUz dxnSOi"
            >
              <img
                alt="front_default"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/168.png"
              />
            </div>
          </styled.div>
        </div>
        <div
          key="front_shiny"
        >
          <styled.div>
            <div
              className="sc-gsTCUz dxnSOi"
            >
              <img
                alt="front_shiny"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/168.png"
              />
            </div>
          </styled.div>
        </div>
      </Suspense>
    </div>
  </styled.div>
</Sprites>
`;
